lib/_tls_common.js

```diff
@@ -132,6 +133,49 @@ exports.createSecureContext = function createSecureContext(options, context) {
     }
   }
 
+  if (options.pskCallback && c.context.enablePskCallback) {
+    c.context.onpskexchange = (identity, maxPskLen, maxIdentLen) => {
+      let ret = options.pskCallback(identity);
+      if (typeof ret !== 'object') {
+        ret = undefined;
+      }
+
+      if (ret) {
+        ret = { psk: ret.psk, identity: ret.identity };
+
+        if (!typeof ret.psk === 'string') {
+          throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'psk', ['string']);
+        }
+        if (ret.psk.length / 2 > maxPskLen) {
+          throw new errors.RangeError(
+            'ERR_ASSERTION',
+            `Pre-shared key exceed ${maxPskLen} bytes`
+          );
+        }
+
+        // Only set for the client callback, which must provide an identity.
+        if (maxIdentLen) {
+          if (typeof ret.identity !== 'string') {
+            throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'identity',
+                                       'string');
+          }
+          if (ret.identity.length > maxIdentLen) {
+            throw new errors.RangeError(
+              'ERR_ASSERTION',
+              `PSK identity exceeds ${maxIdentLen} bytes`
+            );
+          }
+        }
+      }
+
+      return ret;
+    };
+    c.context.enablePskCallback();
```

Comment:
If I'm reading the `#ifdef` statements write in the source below, there's a chance the `enablePskCallback()` will not be defined if `OPENSSL_PSK_SUPPORT` is switched off, correct? I may be missing something because I haven't done a complete walkthrough, but if that's the case, I'd recommend gating these with a flag in `node_config.cc`

Reply:
There shouldn't be a chance that `enablePskCallback` is enabled, since the definition of that is also inside of an `#ifdef`, see [here](https://github.com/nodejs/node/pull/14978/files/d434fe9287824e325ba11784d5c3f46ebd93dec5#diff-801e3948990f4965a8ea4aca4a423864R343)

Reply:
Ok, I see, I was misreading the location of the closing brackets. This call is made inside the if that checks `c.context.enablePskCallback` is not falsy. :-)